### PR TITLE
Refresh 1.14.5 release notes

### PR DESCRIPTION
**Pull request by rnicoll**
Likely still needs some tweaks, but this should be 80% of the way there at least.

### CODE REVIEW

File: doc/release-notes.md [hunk 1]
```diff
@@ -65,13 +80,43 @@ are consistent, so that extended keys from one can be used with the other. This
...
+* Add xkbcommon 0.8.4 to fix keyboard compatibility issues.
+* Address compatibility issues with GCC-7 and glibc-2.27.
+* Remove legacy patches which are no longer needed due to test environments being updated to more recent Ubuntu releases.
+* Security harden systemd unit file `contrib/init/dogecoind.local.service`.
+* Make Freetype library version independent from build system libaries.
+* Update Univalue library to 1.0.4.
+* Correct block download timeout for regtest, where it was too low and causing issues with tests.
+* Experimental build for SHA algorithms with AVX2 support, to improve SHA performance.
 
 Credits
```
Comment (patricklodder) [inline doc/release-notes.md:121]: you're missing some authors in co-author lines in squashed commits:

```
* AbcSxyZ
* Bertrand Jacquin
* Carl Dong
* cg
* chromatic
* Chun Kuan Lee
* Cory Fields
* Dakoda Greaves
* Daksh Sharma
* Dan Raviv
* dogespacewizard
* Ed Tubbs
* Elvis Begović
* fanquake
* Hennadii Stepanov
* KabDeveloper
* leezhen
* Micael Malta
* Michi Lumin
* Patrick Lodder
* Ross Nicoll
* Ryan Crosby
* Suhas Daftuar
* W. J. van der Laan
```

**Review by patricklodder — CHANGES_REQUESTED** [review]
I was working on this in parallel earlier. I'm okay with this update, except you're missing some credits, see comment below.

I have some rewrites around fees and dust that I'd like to propose, but would prefer to pull that in on top of this.

File: doc/release-notes.md (hunk 1 above)
Comment (rnicoll) [reply doc/release-notes.md:123]: Done, thanks

**Review by patricklodder — APPROVED** [review]
ACK. will propose reorg / rewording separately.

**Pull request merged**